fix(message-input): DLT-2605 fix slash command and close button

braddialpad opened the pull request:
# fix(message-input): DLT-2605 fix slash command and close button

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![Obligatory GIF](https://media1.giphy.com/media/v1.Y2lkPTc5MGI3NjExaTJqZXlhNHVtcGVkNWtrM3FzMnUwcDc3eTY5b25pMTczb3B1azZzdCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/ytlPLqdKk0Df2/giphy.gif)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

These types will increment the version number on release:

- [x] Fix

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-2605

## :book: Description

- Fixed slash commands not showing up in storybook (value vs modelValue issue)
- Fixed meeting pill close button not triggering event (cannot use $parent in vue 3)

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have added / updated unit tests.
- [x] I have validated components with a screen reader.
- [x] I have validated components keyboard navigation.

## :crystal_ball: Next Steps

Verify in product

## :camera: Screenshots / GIFs

![Screenshot 2025-06-26 at 12 31 08 PM](https://github.com/user-attachments/assets/933063bc-ffd2-40e6-9a08-aa8a1cd6294b)

## :link: Sources

https://tiptap.dev/docs/editor/extensions/custom-extensions/extend-existing#storage

josedialpad commented on docs/components/avatar.md:
```diff
@@ -10,19 +10,29 @@ figma_url: https://www.figma.com/file/2adf7JhZOncRyjYiy2joil/DT-Core%3A-Componen
         <div class="d-avatar d-avatar--md">
-            <svg aria-hidden="true" focusable="false" data-name="Video" class="d-icon d-icon--video d-icon--size-300" viewBox="0 0 12 12" data-v-68a9d516=""><path fill="currentColor" fill-rule="evenodd" d="M2 3.5a.5.5 0 0 0-.5.5v4a.5.5 0 0 0 .5.5h5a.5.5 0 0 0 .5-.5V4a.5.5 0 0 0-.5-.5H2Zm6.5 1.566V4A1.5 1.5 0 0 0 7 2.5H2A1.5 1.5 0 0 0 .5 4v4A1.5 1.5 0 0 0 2 9.5h5A1.5 1.5 0 0 0 8.5 8V6.934l2.223 1. ...
+            <div class="d-avatar__canvas">
+                <span class="d-avatar__icon">
+                    <span class="d-avatar__icon">
+                        <dt-icon name="video" size="300" />
+                    </span>
+                </span>
```
Remove extra `d-avatar__icon` wrapper

```suggestion
                <span class="d-avatar__icon">
                      <dt-icon name="video" size="300" />
                </span>
```

francisrupert replied:
Good catch.

josedialpad commented on docs/components/avatar.md:
```diff
@@ -221,6 +377,40 @@ Unique gradient combinations are generated by the Vue component. It randomizes s
 
+### Group
+
+<code-well-header>
```
Can be useful to exemplify the usage of the group count with the different sizes as it's in the presence example?

francisrupert replied:
One size only as acknowledged this morning.

josedialpad reviewed (commented):
Great work, approved w/ 2 suggestions. Will give the formal approval when the vue implementation is updated as this avatar [here](https://dialpad.atlassian.net/browse/DT-891) as this update contains some breaking changes.

francisrupert commented:
@josedialpad I see that you've got [DT-891](https://dialpad.atlassian.net/browse/DT-891) in progress. I'll defer to you and @braddialpad for timing the approval and merge of this to staging as it relates to timing the release of both `dialtone` and `dialtone-vue`.

josedialpad commented:
> @josedialpad I see that you've got [DT-891](https://dialpad.atlassian.net/browse/DT-891) in progress. I'll defer to you and @braddialpad for timing the approval and merge of this to staging as it relates to timing the release of both `dialtone` and `dialtone-vue`.

Yeah, created a [draft PR](https://github.com/dialpad/dialtone-vue/pull/745) for the vue changes, once the vue2 and vue3 PRs are approved we can move forward with merging and releasing.

francisrupert commented:
@josedialpad [21fc2ed](https://github.com/dialpad/dialtone/pull/813/commits/21fc2ed6801e94ec3213da92914d259c4d72b1dc) resolves.

francisrupert commented:
@braddialpad Yeah, that would be wise.

braddialpad commented:
Blog post PR up here https://github.com/dialpad/dialtone/pull/816

braddialpad reviewed (approved):
Jose's PR is now approved this is good to merge.

braddialpad committed:
fix(message_input): DLT-2605 fix slash command and close button

ninamarina commented:
I am still getting the `Cannot read properties of null (reading '$emit')` error when using a slash command in the preview:
<img width="1152" alt="image" src="https://github.com/user-attachments/assets/cd0ab118-0a8f-4e03-a6fc-81864246f363" />

braddialpad commented:
> I am still getting the `Cannot read properties of null (reading '$emit')` error when using a slash command in the preview:

Ah yes, we have the same issue in the "SlashCommandComponent" extension..

braddialpad committed:
fix slash_command extension also

braddialpad committed:
add tests

braddialpad committed:
migrate to vue 2 also

braddialpad commented:
Fixed the slash command error and added tests for both the SlashCommand and MeetingPill extensions. Also migrated to vue 2 since this method keeps it in sync much better between the two version since it gets rid of `$parent`.

@ninamarina

braddialpad commented:
Seems like theres some problems with setting the initial text on percy vue 2. Will look into on monday.

braddialpad committed:
fix modelValue on vue2

braddialpad commented:
Percy failure fixed.
